src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -90,18 +76,39 @@ static MsQuicApi()
                         if (StatusSucceeded(apiTable->GetParam(null, QUIC_PARAM_GLOBAL_LIBRARY_VERSION, &size, libVersion)))
                         {
                             var version = new Version((int)libVersion[0], (int)libVersion[1], (int)libVersion[2], (int)libVersion[3]);
-                            if (version >= MsQuicVersion)
-                            {
-                                Api = new MsQuicApi(apiTable);
-                                IsQuicSupported = true;
-                            }
-                            else
+                            if (version < MsQuicVersion)
                             {
                                 if (NetEventSource.Log.IsEnabled())
                                 {
                                     NetEventSource.Info(null, $"Incompatible MsQuic library version '{version}', expecting '{MsQuicVersion}'");
                                 }
+                                return;
+                            }
+
+                            // Assume SChanel is being used on windows and query for the actual provider from the library
+                            QUIC_TLS_PROVIDER provider = OperatingSystem.IsWindows() ? QUIC_TLS_PROVIDER.SCHANNEL : QUIC_TLS_PROVIDER.OPENSSL;
+                            size = sizeof(QUIC_TLS_PROVIDER);
+                            apiTable->GetParam(null, QUIC_PARAM_GLOBAL_TLS_PROVIDER, &size, &provider);
+                            UsesSChannelBackend = provider == QUIC_TLS_PROVIDER.SCHANNEL;
+
+                            if (UsesSChannelBackend)
+                            {
+                                if (!IsWindowsVersionSupported())
+                                {
+                                    if (NetEventSource.Log.IsEnabled())
+                                    {
+                                        NetEventSource.Info(null, $"Current Windows version ({Environment.OSVersion}) is not supported by QUIC. Minimal supported version is {MinWindowsVersion}");
+                                    }
+
+                                    return;
+                                }
+
+                                Tls13ServerMayBeDisabled = IsTls13Disabled(true);
+                                Tls13ClientMayBeDisabled = IsTls13Disabled(false);
```

Comment:
```suggestion
                                Tls13ServerMayBeDisabled = IsTls13Disabled(isServer: true);
                                Tls13ClientMayBeDisabled = IsTls13Disabled(isServer: false);
```
I know you're just moving it and not changing, but it's always nice to have a name of parameter with the boolean constants.